pkg/scalers/nats_jetstream_scaler.go

```diff
@@ -0,0 +1,271 @@
+	State     streamState      `json:"state"`
+	Consumers []consumerDetail `json:"consumer_detail"`
+}
+
+type streamConfig struct {
+	Subjects []string `json:"subjects"`
+}
+
+type streamState struct {
+	MsgCount     int64 `json:"messages"`
+	LastSequence int64 `json:"last_seq"`
+}
+
+type consumerDetail struct {
+	StreamName     string                 `json:"stream_name"`
+	Name           string                 `json:"name"`
+	NumAckPending  int                    `json:"num_ack_pending"`
+	NumRedelivered int                    `json:"num_redelivered"`
+	NumWaiting     int                    `json:"num_waiting"`
+	NumPending     int                    `json:"num_pending"`
+	Config         consumerConfig         `json:"config"`
+	DeliveryStatus consumerDeliveryStatus `json:"delivery"`
+}
+
+type consumerConfig struct {
+	DurableName string `json:"durable_name"`
+}
+
+type consumerDeliveryStatus struct {
+	ConsumerSequence int64 `json:"customer_seq"`
+	StreamSequence   int64 `json:"stream_seq"`
+}
+
+func NewNATSJetStreamScaler(config *ScalerConfig) (Scaler, error) {
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
+	}
+
+	jsMetadata, err := parseNATSJetStreamMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing NATS JetStream metadata: %s", err)
+	}
+
+	return &natsJetStreamScaler{
+		metricType: metricType,
+		stream:     &streamDetail{},
+		metadata:   jsMetadata,
+		httpClient: kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, false),
+	}, nil
+}
+
+func parseNATSJetStreamMetadata(config *ScalerConfig) (natsJetStreamMetadata, error) {
+	meta := natsJetStreamMetadata{}
+	var err error
+	meta.monitoringEndpoint, err = GetFromAuthOrMeta(config, "natsServerMonitoringEndpoint")
+	if err != nil {
+		return meta, err
+	}
+
+	if config.TriggerMetadata["account"] == "" {
+		return meta, errors.New("no account name given")
+	}
+	meta.account = config.TriggerMetadata["account"]
+
+	if config.TriggerMetadata["stream"] == "" {
+		return meta, errors.New("no stream name given")
+	}
+	meta.stream = config.TriggerMetadata["stream"]
+
+	if config.TriggerMetadata["consumer"] == "" {
+		return meta, errors.New("no consumer name given")
+	}
+	meta.consumer = config.TriggerMetadata["consumer"]
+
+	meta.lagThreshold = defaultJetStreamLagThreshold
+
+	if val, ok := config.TriggerMetadata[lagThresholdMetricName]; ok {
+		t, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return meta, fmt.Errorf("error parsing %s: %s", lagThresholdMetricName, err)
+		}
+		meta.lagThreshold = t
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+	return meta, nil
+}
+
+func (s *natsJetStreamScaler) getNATSJetStreamEndpoint() string {
+	return fmt.Sprintf("http://%s/jsz?acc=%s&consumers=true&config=true", s.metadata.monitoringEndpoint, s.metadata.account)
+}
+
+func (s *natsJetStreamScaler) IsActive(ctx context.Context) (bool, error) {
```

Comment:
We have recently introduced configurable activation, could you please add it here: https://github.com/kedacore/keda/issues/2800